bind_slurm.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Specify system-specific bindings to use here.
+
+CONFIG=nesh.conf
+
+
+
+# Load bindings from config.
+
+source ${CONFIG}
+
+echo Will use: ${CONFIG}
+
+# Check if singularity is available.
+
+singularity --version || { echo Singularity not found... exiting.; exit; }
+
+# Get Singularity image path.
+
+ORIGINAL_SINGULARITY_COMMAND="$@" # Preserve original argument
+
+echo Will execute: ${ORIGINAL_SINGULARITY_COMMAND}
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    *.sif) IMAGE="$1" ;;
+    *) ;;
+  esac
+  shift
+done
+
+echo Enable host SLURM user for: ${IMAGE}
+
+# Prepare system-specific SLURM user binding.
+
+TEMPDIR=$(mktemp -d $(pwd)/tmp.XXXXXXXX)
+echo Temporary directory: ${TEMPDIR}
+trap "echo Deleting: ${TEMPDIR}; rm -rf ${TEMPDIR}" 0
+
+singularity exec ${IMAGE} cp -p /etc/passwd ${TEMPDIR}/etc_passwd
+singularity exec ${IMAGE} cp -p /etc/group ${TEMPDIR}/etc_group
+cat /etc/passwd | grep slurm >> ${TEMPDIR}/etc_passwd
+cat /etc/group | grep slurm >> ${TEMPDIR}/etc_group
+
+#getent passwd >> ${TEMPDIR}/etc_passwd
+#getent group >> ${TEMPDIR}/etc_group
+
+RTEMPDIR=$(basename $TEMPDIR) # get relative temp dir path
+
+MERGED_PASSWD_GROUP="\
+${RTEMPDIR}/etc_passwd:/etc/passwd,\
+${RTEMPDIR}/etc_group:/etc/group\
+"
+
+# Setup Singularity bind mount and (library) path environment.
+
+if [[ ! -z "$SLURM_PREPEND_PATH" ]]; then
+export SINGULARITYENV_PREPEND_PATH=${SLURM_PREPEND_PATH}${SINGULARITYENV_PREPEND_PATH:+:$SINGULARITYENV_PREPEND_PATH}
+echo SINGULARITYENV_PREPEND_PATH: "${SINGULARITYENV_PREPEND_PATH}"
+fi
```

Comment:
```suggestion
if [[ ! -z "$BATCH_PREPEND_PATH" ]]; then
export SINGULARITYENV_PREPEND_PATH=${BATCH_PREPEND_PATH}${SINGULARITYENV_PREPEND_PATH:+:$SINGULARITYENV_PREPEND_PATH}
echo SINGULARITYENV_PREPEND_PATH: "${SINGULARITYENV_PREPEND_PATH}"
fi
```

Reply:
I will go for `SCHEDULER_PREPEND_PATH` in a separate commit.